drivers/timer/riscv_opensbi_timer.c

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright (c) 2024 sensry.io
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdint.h>
+#include <limits.h>
+
+#include <zephyr/init.h>
+#include <zephyr/drivers/timer/system_timer.h>
+#include <zephyr/spinlock.h>
+#include <zephyr/irq.h>
+#include <opensbi.h>
+
+static struct k_spinlock lock;
+static uint64_t last_count;
+static uint64_t last_ticks;
+static uint32_t last_elapsed;
+
+#define TIMER_IRQN		0x5
```

Comment:
IRQn should be retrieved from the devicetree

Reply:
agree, however there is no device defined for sys_clock so I m wondering where to store the IRQN properly. should I create a device for sys_clock?